### PR TITLE
Replace query-string with `URLSearchParams` in embed iframe generation

### PR DESCRIPTION
**Pull request by robertknight**
Per the request in https://github.com/hypothesis/client/pull/3589#pullrequestreview-710534189, this PR extracts a subset of the changes from https://github.com/hypothesis/client/pull/3589 for easier review.

It replaces the use of the query-string dependency with URLSearchParams in `sidebar/media-embedder.js`.

### CODE REVIEW

File: src/sidebar/media-embedder.js [hunk 1]
```diff
@@ -103,29 +101,33 @@ function parseTimeString(timeValue) {
...
+      // `t` accepts more formats than `start`; start must be in seconds
+      // so, format it as seconds first
+      filteredQuery.append('start', parseTimeString(value));
+    } else {
+      filteredQuery.append(key, value);
+    }
+  }
+
+  // Tests currently expect sorted parameters.
+  filteredQuery.sort();
```
Comment (robertknight) [inline src/sidebar/media-embedder.js:128]: I could have removed this and revised the test expectations, but it really doesn't matter at present.

File: src/tsconfig.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "allowJs": true,
     "checkJs": true,
-    "lib": ["es2018", "dom"],
+    "lib": ["es2018", "dom", "dom.iterable"],
```
Comment (lyzadanger) [inline src/tsconfig.json:5]: Curious, is https://github.com/microsoft/TypeScript/tree/main/lib a reasonable source of available `lib`s, or is there a more useful source? What was your process of working out that this was what was needed to support `URLQueryParams` (which is iterable)?

File: src/sidebar/media-embedder.js [hunk 3]
```diff
@@ -103,29 +101,33 @@ function parseTimeString(timeValue) {
...
+
+  let query = filteredQuery.toString();
```
Comment (lyzadanger) [inline src/sidebar/media-embedder.js:130]: This pre-exists these changes, but the documented `@returns` for this function doesn't capture the possibility of the returned string being empty (if none of the parameters made it through filtering).

**Review by lyzadanger — APPROVED** [review]
Thanks for breaking these up! Gives me good focus as a reviewer.

File: src/tsconfig.json (hunk 2 above)
Comment (robertknight) [reply src/tsconfig.json:5]: I got confused at first and then googled for the solution, which pointed me to an issue in the TypeScript repo on GitHub.

**Pull request merged**